hd-wallet/pluginv2.lua

```diff
@@ -0,0 +1,511 @@
+    return sobject
+end
+-- export secret key that holds the BIP32 master key --
+-- export BIP32 master key from SDKMS ---
+function export_secret_key(keyId)
+    return Sobject { kid = keyId }:export().value
+end
+
+---------- @@@@@@@@@@@@@@@@@@@@@@@@@ ----------
+----------           UTXO            ----------
+---------- @@@@@@@@@@@@@@@@@@@@@@@@@ ----------
+
+---------- integer to bytearry impl -----------
+---------- utils methods ----------------------
+local random = math.random
+-- convert number into hex string 
+function num_2_hex(num, size)
+    local hexstr = '0123456789ABCDEF'
+    local s = ""
+    while num > 0 do
+        local mod = math.fmod(num, 16)
+        s = string.sub(hexstr, mod+1, mod+1) .. s
+        num = math.floor(num / 16)
+    end
+    if (string.len(s) < size) then
+        local offset = string.rep("0", size-string.len(s))
+        s = offset..s
+    end
+    return s
+end
+---------- @@@@@@@@@@@@@@@@@@@@@@@@@ ----------
+------------- Parse Derivation Path -----------
+---------- @@@@@@@@@@@@@@@@@@@@@@@@@ ----------
+function split(str, pat)
+    local t = {}
+    local fpat = "(.-)" .. pat
+    local last_end = 1
+    local s, e, cap = str:find(fpat, 1)
+    while s do
+        if s ~= 1 or cap ~= "" then
+            table.insert(t,cap)
+        end
+        last_end = e+1
+        s, e, cap = str:find(fpat, last_end)
+    end
+    if last_end <= #str then
+        cap = str:sub(last_end)
+        table.insert(t, cap)
+    end
+    return t
+end
+-- extract co-ordinate from complate ec public key
+-- first half of last 64 bit is x-cordinate and second half is y-cordinate
+function extract_coordinates_from_ASN1_public_key(key_byte)
+    return string.sub(key_byte, 49, 176)
+end
+-- return ec curve co-ordinate from private key
+function get_point_coordinates_from_private_key(key_byte)
+    local asn1_ec_key = create_ASN1_private_key(key_byte)
+    local blob = Blob.from_hex(asn1_ec_key)
+    local ec_key = import_ec_key(blob)
+    local asn1_public_key = ec_key.pub_key:hex()
+    local coordinate = extract_coordinates_from_ASN1_public_key(asn1_public_key)
+    return coordinate
+end
+-- compress key co-ordinate
+function compress_public_key(x, y)
+    local a = BigNum.from_bytes_be(Blob.from_hex(y))
+    local b = BigNum.from_bytes_be(Blob.from_hex("02"))
+    local c = BigNum.from_bytes_be(Blob.from_hex("00"))
+
+    if (a % b):to_bytes_be() == c:to_bytes_be() then
+        return "02"..x
+    else 
+        return "03"..x
+    end 
+end
+-- return public key from private key 
+function public_key_for_private_key(key_byte)
+    local point = get_point_coordinates_from_private_key(key_byte)
+    return compress_public_key(string.sub(point, 1, 64), string.sub(point, 65, 128))
+end
+-- parse input path
+function parse_path(child_path)
+    local path_table = split(child_path, "/")
+    return path_table
+end
+-- import chain-code as hmac key
+-- sign data as from hmac key
+function get_hmac(hmac_key, data)
```

Comment:
This two-line function is only used once, makes sense to put the 2 lines there.

Reply:
Agreed, done